### ISSUE TITLE
In dev mode on vite using remix I don't get the listener cleanup functions called

### ISSUE THREAD

**Issue by vidjuheffex** [baggers-org/remix-sse#27]
In prod mode everything works fine.

However, in dev mode on vite using remix I don't get the listener cleanup functions called. HMR?

**Comment by dan-cooke** [comment]
@vidjuheffex yes its most likely an issue with HMR, i will look into it when I get a chance

**Comment by larswww** [comment]
Evaluating libraries for SSE and saw that remix-utils have a similar issue in their implementation. Perhaps you'll find this thread helpful: https://github.com/sergiodxa/remix-utils/issues/317

**Comment by mkromann** [comment]
This is still an issue as far as I can tell?

Does anybody have a workaround for how to deal with this? Currently it seems infeasible to develop with SSE and Remix, as any complex SSE bogs down in development.
